miner/worker_builder.go

```diff
@@ -387,6 +398,11 @@ func (w *worker) simulateBundle(
 		}
 
 		if receipt.Status == types.ReceiptStatusFailed && !containsHash(bundle.RevertingTxHashes, receipt.TxHash) {
+			// for unRevertible tx but itself can be dropped, we drop it and revert the state and gas pool
+			if containsHash(bundle.DroppingTxHashes, receipt.TxHash) {
+				log.Warn("drop tx in bundle", "hash", receipt.TxHash.String())
```

Comment:
We should also do this here?
```
state.RevertToSnapshot(snap)
gasPool.SetGas(gp)
```

Reply:
We can't do this here beacause in the evm execution, when status was failed, it was already reverted. If revert here, it will panic.

Reply:
The actual reason for no need to revert state here is that, when no err returned by ApplyTransaction, state.clearJournalAndRefund() must had been called by  https://github.com/bnb-chain/op-geth/blob/develop/core/state_processor.go#L167 or https://github.com/bnb-chain/op-geth/blob/develop/core/state/statedb.go#L1008 to avoid reverting across transactions, after that, `state.validRevisions` would be clear, so if here do revert, panic must happen for the first check is true at https://github.com/bnb-chain/op-geth/blob/develop/core/state/statedb.go#L918, so we directly remove the tx from bundle.